source/d3d8to9_base.cpp

```diff
@@ -10,10 +10,11 @@ static const D3DFORMAT AdapterFormats[] = {
 	D3DFMT_X8R8G8B8,
 	D3DFMT_R5G6B5,
 	D3DFMT_X1R5G5B5,
-	D3DFMT_A1R5G5B5,
-	D3DFMT_A2R10G10B10
+	D3DFMT_A1R5G5B5
 };
 
+IDirect3DDevice9 *pCurrentDeviceInterface = nullptr;
```

Comment:
@crosire, Yes, I tried using `Direct3D8` class here first.  It still [crashed](https://github.com/crosire/d3d8to9/issues/39#issuecomment-322025422) after this because GTA VC would call `Release()` and release the old device after we had already released it.  If we release the `Direct3D8` device we will delete it see [here](https://github.com/crosire/d3d8to9/blob/fc2dcac3531bcda61f48dcf0566eb04559832370/source/d3d8to9_base.cpp#L82) which causes GTA VC to crash.  However the `IDirect3DDevice9` pointer is always available even after it is released so GTA VC will not crash if we release `IDirect3DDevice9` but will crash if we release `Direct3D8`.  I also tested both ways and was able to reproduce the crash (after buying the game from Steam).  This may be an issue only with the Steam version because when I used a hacked version (so I did not have to buy the game) it did not crash here.

@CookiePLMonster, Yes, I think this is a bug in GTA.  That is why I commented above and said "GTA does not know how to properly handle this error and simply crashed."

BTW: I wonder if there is a way to do the same thing that Direct3D 9 does here and keep the pointer available even after we delete the class?  This way the game would not crash and we would not need to do caching like we did for [Need for Speed III](https://github.com/crosire/d3d8to9/pull/20).